src/main/java/friday/Friday.java

```diff
@@ -0,0 +1,148 @@
+                    t.markAsDone();
+                    storage.save(tasks.asList());
+                    System.out.println(separator);
+                    System.out.println("Okay boss, I have marked this task as done:");
+                    System.out.println(" " + t);
+                    System.out.println(separator);
+                } else if (command.startsWith("unmark")) {
+                    String[] parts = Parser.splitOnce(command, "\\s+");
+                    if (parts.length < 2) {
+                        throw new FridayException("Apologies boss, that task number isn't recorded in my database.");
+                    }
+                    int index = Integer.parseInt(parts[1]) - 1;
+                    Task t = tasks.get(index);
+                    t.markAsNotDone();
+                    storage.save(tasks.asList());
+                    System.out.println(separator);
+                    System.out.println("Okay boss, I have marked this task as not done yet:");
+                    System.out.println(" " + t);
+                    System.out.println(separator);
+                } else if (command.startsWith("todo")) {
+                    String desc = command.length() > 4 ? command.substring(4).trim() : "";
+                    if (desc.isEmpty()) {
+                        throw new FridayException("Seems like you have made a mistake boss, a todo cannot be empty.");
+                    }
+                    Task t = new Todo(desc);
+                    tasks.add(t);
+                    storage.save(tasks.asList());
+                    System.out.println(separator);
+                    System.out.println("Got it boss. I have added this task:");
+                    System.out.println(" " + t);
+                    System.out.println("Boss, you have " + tasks.size() + " tasks in the list.");
+                    System.out.println(separator);
+                } else if (command.startsWith("deadline")) {
+                    String details = command.length() > 8 ? command.substring(8).trim() : "";
+                    String[] parts = Parser.splitOnce(details, " /by ");
+                    if (parts.length < 2 || parts[0].trim().isEmpty() || parts[1].trim().isEmpty()) {
+                        throw new FridayException("Seems like you have made a mistake boss, a deadline cannot be empty.");
+                    }
+                    String desc = parts[0].trim();
+                    String by = parts[1].trim();
+                    Deadline t = new Deadline(desc, by);
+                    tasks.add(t);
+                    storage.save(tasks.asList());
+                    printAddMessage(t, tasks.size(), separator);
+                } else if (command.startsWith("event")) {
+                    String details = command.length() > 5 ? command.substring(5).trim() : "";
+                    int fromIdx = Parser.indexOf(details, " /from ");
+                    int toIdx = Parser.indexOf(details, " /to ");
```

Comment:
Nit: I think the code quality guidelines mention to avoid 'texting-style' spelling, would it be better to spell out Index instead?

```suggestion
                    int fromIndex = Parser.indexOf(details, " /from ");
                    int toIndex = Parser.indexOf(details, " /to ");
```